src/Configuration/src/SpringBoot/SpringBootCommandLineProvider.cs

```diff
@@ -39,8 +40,7 @@ public override void Load()
 
             if (key.StartsWith(KeyPrefix, StringComparison.OrdinalIgnoreCase))
             {
-                string newKey = key.Replace('.', ':');
-                Data[newKey] = pair.Value;
+                Data[ConfigurationKeyConverter.AsDotNetConfigurationKey(key)] = pair.Value;
```

Comment:
Why call the new method here? I'd prefer to isolate usage of the questionable implementation as much as possible, instead of using it everywhere.

Reply:
I disagree. From what we understand so far, the method was created to handle escaped arrays in YAML files that exist in a git repo being served through Config Server. That's a very specific scenario, which doesn't apply to all other places. We haven't actually verified what ConfigServer makes of YAML files, so we may as well be unescaping too much here, or handling the wrong escape sequence. I'd rather isolate that uncertainty to ConfigServer, than to spread it over various places. If YAML files use brackets to escape values, then why does our code only allow numbers between brackets? Does ConfigServer unescape them already? There are many more questions, and it requires more investigation and trying it out to see what Steeltoe needs to handle.

If all the other places are wrong, we should have evidence of that and fix them accordingly. They may just as well need a different escape scheme. Just calling one complex method that's poorly defined (and with concerning performance impact) from everywhere, only in case it might help anyone, may as well work against us. Enhancing a simple conversion as a bugfix is fine, but introducing breaking changes because we fix one place and break others without noticing is not.